src/features/instantiate/containers/InstantiateContainer.tsx

```diff
@@ -15,14 +16,21 @@ export const InstantiateContainer = () => {
   changeBodyBackground(AltinnPalette.greyLight);
   const party = useCurrentParty();
   const instantiation = useInstantiation();
+  const clearRef = useAsRef(instantiation.clear);
 
-  React.useEffect(() => {
-    const shouldCreateInstance = !!party && !instantiation.lastResult && !instantiation.isLoading;
+  useEffect(() => {
+    const shouldCreateInstance = !!party;
     if (shouldCreateInstance) {
       instantiation.instantiate(undefined, party.partyId);
     }
   }, [instantiation, party]);
 
+  // Clear the instantiation when the component is unmounted, to allow users to start a new instance later
+  useEffect(() => {
+    const clear = clearRef.current;
+    return () => clear();
+  }, [clearRef]);
```

Comment:
Do you expect the ref value to change during the lifetime of this component? If so the cleanup function uses the ref value at the time the component mounts not the current ref value at the time it unmounts

Reply:
Yup, it might change, but that's just because `InstantiationContext` doesn't do any memoization. :face_with_peeking_eye: But even if it changes, it shouldn't really matter - the `setState()` call for `busyWithId` updates the same state, and the rest works directly on the queryClient. I had to write it this way because the ref itself may be gone when the cleanup function runs, and just writing a `useEffect()` with the `clear` function as a dependency would clean up everything as soon as anything happens (because of the missing memoization). :shrug: